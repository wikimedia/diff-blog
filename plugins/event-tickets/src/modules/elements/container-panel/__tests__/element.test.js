/**
 * External dependencies
 */
import React from 'react';

/**
 * Internal dependencies
 */
import ContainerPanel, { LAYOUT } from '@moderntribe/tickets/elements/container-panel/element';

describe( 'Container Panel Element', () => {
	it( 'renders container panel in rsvp layout', () => {
		const component = renderer.create( <ContainerPanel layout={ LAYOUT.rsvp } /> );
		expect( component.toJSON() ).toMatchSnapshot();
	} );

	it( 'renders container panel in ticket layout', () => {
		const component = renderer.create( <ContainerPanel layout={ LAYOUT.ticket } /> );
		expect( component.toJSON() ).toMatchSnapshot();
	} );

	it( 'renders container panel with icon, header, and content', () => {
		const props = {
			icon: 'icon',
			header: 'header',
			content: 'content',
			layout: LAYOUT.rsvp,
		};
		const component = renderer.create( <ContainerPanel { ...props } /> );
		expect( component.toJSON() ).toMatchSnapshot();
	} );

	it( 'renders container panel with classes', () => {
		const props = {
			className: 'test-class-name',
			layout: LAYOUT.rsvp,
		};
		const component = renderer.create( <ContainerPanel { ...props } /> );
		expect( component.toJSON() ).toMatchSnapshot();
	} );
} );
