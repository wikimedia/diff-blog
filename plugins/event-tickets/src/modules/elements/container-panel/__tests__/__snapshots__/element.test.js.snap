// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container Panel Element renders container panel in rsvp layout 1`] = `
<div
  className="tribe-editor__container-panel tribe-editor__container-panel--rsvp"
>
  <div
    className="tribe-editor__container-panel__icon"
  />
  <div
    className="tribe-editor__container-panel__header-content-wrapper"
  >
    <div
      className="tribe-editor__container-panel__header"
    />
  </div>
</div>
`;

exports[`Container Panel Element renders container panel in ticket layout 1`] = `
<div
  className="tribe-editor__container-panel tribe-editor__container-panel--ticket"
>
  <div
    className="tribe-editor__container-panel__icon"
  />
  <div
    className="tribe-editor__container-panel__header"
  />
</div>
`;

exports[`Container Panel Element renders container panel with classes 1`] = `
<div
  className="tribe-editor__container-panel tribe-editor__container-panel--rsvp test-class-name"
>
  <div
    className="tribe-editor__container-panel__icon"
  />
  <div
    className="tribe-editor__container-panel__header-content-wrapper"
  >
    <div
      className="tribe-editor__container-panel__header"
    />
  </div>
</div>
`;

exports[`Container Panel Element renders container panel with icon, header, and content 1`] = `
<div
  className="tribe-editor__container-panel tribe-editor__container-panel--rsvp"
>
  <div
    className="tribe-editor__container-panel__icon"
  >
    icon
  </div>
  <div
    className="tribe-editor__container-panel__header-content-wrapper"
  >
    <div
      className="tribe-editor__container-panel__header"
    >
      header
    </div>
    <div
      className="tribe-editor__container-panel__content"
    >
      content
    </div>
  </div>
</div>
`;
